### PR TITLE
Fix Sumo Logic output plugin not splitting requests properly (#25)

### PR DESCRIPTION
**Pull request by pmalek-sumo**
This should fix 2 problems:

* sending the last request with empty body (the condition when calculating boundary was incorrect)
* sending requests bigger than `MaxRequstBodySize` (now it will only happen when it's configured so small so that it's not able to accomodate even one metric - a warning will be printed out in such case but the request will go through).

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Review by ssoroka — CHANGES_REQUESTED** [review]
Once the requests are chunked, if any one succeeds, you need to **log** the error (not return it) and continue to attempt to deliver all the rest of the chunks. This means dropping the ones that don't deliver correctly, as there's no way to partially deliver a batch.

A better approach might be to ensure your batch size is small enough that you don't ever need to chunk, as once you start chunking your only options when a chunk fails to deliver is to: 1. Drop chunks that don't make it and accept the batch as delivered (returning no error), or 2. accept delivering duplicates and return an error, retrying the entire batch.

**Comment by pmalek-sumo** [comment]
> A better approach might be to ensure your batch size is small enough that you don't ever need to chunk

Can you explain a bit more how should that work?

> 1. Drop chunks that don't make it and accept the batch as delivered (returning no error), or 2. accept delivering duplicates and return an error, retrying the entire batch.

I'd rather just do 1.

**Comment by pmalek-sumo** [comment]
I've changed the logic to just log the error when a chunk fails to get sent.

On top of that: I have a patch that checks each request's body additionally to checking the request count, I can add it here or push a separate PR if that would be better.

File: plugins/outputs/sumologic/sumologic_test.go [hunk 1]
```diff
@@ -592,26 +603,35 @@ func TestMaxRequestBodySize(t *testing.T) {
...
 	}
 
 	for _, tt := range testcases {
 		t.Run(tt.name, func(t *testing.T) {
+			var (
+				requestCount int
+				linesCount   int
+			)
+			ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				requestCount++
```
Comment (ssoroka) [inline plugins/outputs/sumologic/sumologic_test.go:620]: might need `atomic.AddInt32` / `atomic.LoadInt32` for these counters. It's a different goroutine, and isn't race safe.

**Review by ssoroka — CHANGES_REQUESTED** [review]
Nearly there. two small change requests.

**Review by ssoroka — APPROVED** [review]

**Pull request merged**